time/src/test/java/net/jqwik/time/api/ZoneOffsetTests.java

```diff
@@ -113,6 +113,41 @@ void betweenSame(@ForAll("offsets") ZoneOffset sameOffset, @ForAll Random random
 
 			}
 
+			@Property(shrinking = ShrinkingMode.OFF)
+			@ExpectFailure(failureType = IllegalArgumentException.class)
+			void betweenNotGeneratedValues(
+					@ForAll("times") LocalTime start,
+					@ForAll("times") LocalTime end,
+					@ForAll boolean startB,
+					@ForAll boolean endB,
+					@ForAll Random random
+			) {
+
+				Assume.that(startB || !start.isAfter(LocalTime.of(12, 0, 0)));
+				Assume.that(endB || !end.isAfter(LocalTime.of(12, 0, 0)));
+				int mul = startB ? 1 : -1;
+				ZoneOffset startOffset = ZoneOffset
+												 .ofHoursMinutesSeconds(mul * start.getHour(), mul * start.getMinute(), mul * start.getSecond());
+				mul = endB ? 1 : -1;
+				ZoneOffset endOffset = ZoneOffset.ofHoursMinutesSeconds(mul * end.getHour(), mul * end.getMinute(), mul * end.getSecond());
+
+				Assume.that(startOffset.getTotalSeconds() <= endOffset.getTotalSeconds());
+
+				Arbitrary<ZoneOffset> offsets = Times.zoneOffsets().between(startOffset, endOffset);
+
+				assertAllGenerated(offsets.generator(1000), random, offset -> {
+					assertThat(offset.getTotalSeconds()).isGreaterThanOrEqualTo(startOffset.getTotalSeconds());
+					assertThat(offset.getTotalSeconds()).isLessThanOrEqualTo(endOffset.getTotalSeconds());
+					return true;
+				});
+
+			}
```

Comment:
This testcase looks very complicated, but it is definitely neccessary. It founds two bugs!

Reply:
:-) what do startB and endB stand for?

Reply:
I forgot to rename it. It says whether the offset is negative or positive.